agentuity/server/response.py

```
@@ -54,41 +55,90 @@ def metadata(self) -> dict:
+            return self
+
+        params = self._handoff_params["params"]
+        args = self._handoff_params["args"]
+        metadata = self._handoff_params["metadata"]
+        agent_id = params.get('id') or params.get('name')
+
+        # Enhanced error handling for agent resolution
+        try:
+            found_agent = resolve_agent(self._context, params)
+        except ValueError as e:
+            raise ValueError(f"Handoff failed: Agent '{agent_id}' not found or not accessible. {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)